pay-api/src/pay_api/services/payment_service.py

```diff
@@ -114,31 +117,31 @@ def create_payment(cls, payment_request: Tuple[Dict[str, Any]], authorization: T
                                                             corp_type_code=invoice.corp_type_code)
 
             current_app.logger.debug('Updating invoice record')
-            invoice = Invoice.find_by_id(invoice.id, skip_auth_check=True)
-            invoice.invoice_status_code = pay_service.get_default_invoice_status()
-            invoice.flush()
-            InvoiceReference.create(invoice.id, pay_system_invoice.get('invoice_number', None),
-                                    pay_system_invoice.get('reference_number', None))
+            invoice_reference: InvoiceReference = InvoiceReference.\
+                create(invoice.id, pay_system_invoice.get('invoice_number', None),
+                       pay_system_invoice.get('reference_number', None))
 
-            payment.commit()
-            pay_service.complete_post_payment(payment.id)
-            payment = Payment.find_by_id(payment.id, skip_auth_check=True)
+            invoice.commit()
+
+            pay_service.complete_post_invoice(invoice.id, invoice_reference)
+
+            invoice = Invoice.find_by_id(invoice.id, skip_auth_check=True)
 
         except Exception as e:
             current_app.logger.error('Rolling back as error occured!')
             current_app.logger.error(e)
-            if payment:
-                payment.rollback()
+            if invoice:
```

Comment:
do we have to rollback InvoiceReference?

Reply:
It will rollback all transactions actually, may be should call db.rollback(), invoice.rollback() does a session rollback